InvenTree/plugin/base/barcodes/mixins.py

```diff
@@ -37,3 +51,277 @@ def scan(self, barcode_data):
+                return {"error": _(f"Failed to find placed purchase order for '{order_number}'")}
+
+        if quantity and not isinstance(quantity, int):
+            try:
+                quantity = int(quantity)
+            except ValueError:
+                logger.warning("Failed to parse quantity '%s'", quantity)
+                quantity = None
+
+        #  find incomplete line_items that match the supplier_part
+        line_items = purchase_order.lines.filter(
+            part=supplier_part.pk, quantity__gt=F("received"))
+        if len(line_items) == 1 or not quantity:
+            line_item = line_items[0]
+        else:
+            # if there are multiple line items and the barcode contains a quantity:
+            # 1. return the first line_item where line_item.quantity == quantity
+            # 2. return the first line_item where line_item.quantity > quantity
+            # 3. return the first line_item
+            for line_item in line_items:
+                if line_item.quantity == quantity:
+                    break
+            else:
+                for line_item in line_items:
+                    if line_item.quantity > quantity:
+                        break
+                else:
+                    line_item = line_items.first()
+
+        if not line_item:
+            return {"error": _("Failed to find pending line item for supplier part")}
+
+        no_stock_locations = False
+        if not location:
+            # try to guess the destination were the stock_part should go
+            # 1. check if it's defined on the line_item
+            # 2. check if it's defined on the part
+            # 3. check if there's 1 or 0 stock locations defined in InvenTree
+            #    -> assume all stock is going into that location (or no location)
+            if location := line_item.destination:
+                pass
+            elif location := supplier_part.part.get_default_location():
+                pass
+            elif StockLocation.objects.count() <= 1:
+                if not (location := StockLocation.objects.first()):
+                    no_stock_locations = True
+
+        response = {
+            "lineitem": {
+                "pk": line_item.pk,
+                "purchase_order": purchase_order.pk,
+            }
+        }
+
+        if quantity:
+            response["lineitem"]["quantity"] = quantity
+        if location:
+            response["lineitem"]["location"] = location.pk
+
+        # if either the quantity is missing or no location is defined/found
+        # -> return the line_item found, so the client can gather the missing
+        #    information and complete the action with an 'api-po-receive' call
+        if not quantity or (not location and not no_stock_locations):
+            response["action_required"] = _("Further information required to receive line item")
+            return response
+
+        purchase_order.receive_line_item(
```

Comment:
I would wrap this in a `try/except` block - it may fail due to some reason, and you want to be able to throw the right error message back to the user

Reply:
Please wrap `receive_line_item` method here in a try/except

Reply:
@30350n can you address this? We should handle any errors which occur during `receive_line_item`